include/xtensor/xtiny.hpp

```diff
@@ -0,0 +1,1999 @@
+    {
+        return &m_data[0];
+    }
+
+    template <class V, index_t N, class R>
+    constexpr inline auto
+    xtiny_impl<V, N, R>::size() const -> size_type
+    {
+        return N;
+    }
+
+    template <class V, index_t N, class R>
+    constexpr inline auto
+    xtiny_impl<V, N, R>::max_size() const -> size_type
+    {
+        return N;
+    }
+
+    template <class V, index_t N, class R>
+    constexpr inline auto
+    xtiny_impl<V, N, R>::capacity() const -> size_type
+    {
+        return N;
+    }
+
+    template <class V, index_t N, class R>
+    constexpr inline auto
+    xtiny_impl<V, N, R>::shape() const -> index_type
+    {
+        return {N};
+    }
+
+    template <class V, index_t N, class R>
+    inline auto
+    xtiny_impl<V, N, R>::begin() -> iterator
+    {
+        return m_data;
+    }
+
+    template <class V, index_t N, class R>
+    constexpr inline auto
+    xtiny_impl<V, N, R>::begin() const -> const_iterator
+    {
+        return m_data;
+    }
+
+    template <class V, index_t N, class R>
+    inline auto
+    xtiny_impl<V, N, R>::rbegin() -> reverse_iterator
+    {
+        return reverse_iterator(m_data+N);
+    }
+
+    template <class V, index_t N, class R>
+    constexpr inline auto
+    xtiny_impl<V, N, R>::rbegin() const -> const_reverse_iterator
+    {
+        return const_reverse_iterator(m_data+N);
+    }
+
+    template <class V, index_t N, class R>
+    inline void
+    xtiny_impl<V, N, R>::swap(xtiny_impl & other)
+    {
+        using std::swap;
+        swap(m_data, other.m_data);
+    }
+
+    /************************************************/
+    /* xtiny_impl dynamic shape view implementation */
+    /************************************************/
+
+    template <class V, class R>
+    inline
+    xtiny_impl<V, runtime_size, R>::xtiny_impl()
+    : m_size(0)
+    , m_data()
+    {
+    }
+
+    template <class V, class R>
+    inline
+    xtiny_impl<V, runtime_size, R>::xtiny_impl(representation_type const & begin, representation_type const & end)
+    : m_size(static_cast<size_type>(std::distance(begin, end)))
+    , m_data(begin)
+    {
+    }
+
+    template <class V, class R>
+    template <class IT, class>
+    inline
+    xtiny_impl<V, runtime_size, R>::xtiny_impl(IT begin, IT end)
+    : m_size(static_cast<size_type>(std::distance(begin, end)))
+    , m_data(const_cast<representation_type>(&*begin))
+    {
+    }
+
+    template <class V, class R>
+    inline void
+    xtiny_impl<V, runtime_size, R>::reset(representation_type const & begin, representation_type const & end)
+    {
+        m_size = static_cast<size_type>(std::distance(begin, end));
+        m_data = begin;
+    }
+
+    template <class V, class R>
+    inline void
+    xtiny_impl<V, runtime_size, R>::assign(size_type n, const value_type& v)
+    {
+        XTENSOR_ASSERT_MSG(n == size(), "xtiny_impl::assign(n, v): size mismatch.");
+        for(size_type k=0; k<size(); ++k)
```

Comment:
use `std::fill`?

Reply:
my benchmarks shows little difference between iterator based std::fill vs. for loop:

```
----------------------------------------------------------------
Benchmark                         Time           CPU Iterations
----------------------------------------------------------------
bench_compare_fill/4              2 ns          2 ns  402986315
bench_compare_fill/8              3 ns          3 ns  215037132
bench_compare_fill/64            19 ns         18 ns   36827484
bench_compare_fill/512          141 ns        141 ns    4734883
bench_compare_fill/1000         270 ns        270 ns    2587655
bench_compare_fill2/4             3 ns          3 ns  243319874
bench_compare_fill2/8             3 ns          3 ns  232418915
bench_compare_fill2/64           28 ns         28 ns   24789837
bench_compare_fill2/512         141 ns        141 ns    4891461
bench_compare_fill2/1000        272 ns        271 ns    2550597
```

Reply:
chenged